### PR TITLE
feat: ✨ revamp meetup section

### LINKED ISSUE #60: Finish meet-ups section
1. Eliminate the meet-ups page (for now) and keep the section on the home page
    - Remove the navigation bar link
2. Use the ReactJS Philippines Facebook Page events as our data source
3. Follow the new meet-up section mock-up:
image

https://xd.adobe.com/spec/967e129d-f3bb-4a08-6f39-553da7ddef6d-e04f/

### PR DESCRIPTION
**Pull request by jromest**
## Description
Revamp meetup section based on the new design

## Summary of Changes
- add `date-fns` and `gatsby-source-facebook-graphql` deps
- tweak outline button variant styles
- revamp meetup section based on the new design
- implement gatsby facebook graphql data in meetup section

## How to Test
1. For local testing, add this in the `.env.development` in the project root directory 
  ```
  FACEBOOK_GRAPH_TOKEN=<your-facebook-graph-token>
  ```

2. Generate an access token, be sure you have access to ReactJS PH Facebook page
- Go to https://developers.facebook.com/ and log in using your Facebook account
  - Once you logged in, navigate to **My Apps** on the menu and click **Create App**
  - Give it a display name and your contact email, and click **Create App ID**
- Go to https://developers.facebook.com/tools/explorer/
  - On the Access Token section, select your newly created App in the **Facebook App** dropdown
  - On **User or Page**, select **Get Page Access Token**, a popup will open, select **ReactJS Philippines** and continue with the default config
  - Leave the Permission section as is, and click **Generate Access Token**
- To test your access token on the Graph API Explorer
  - Replace `me?fields=id,name` with `431794124327063/events` and click **Submit**, it must return the events data
  - If all is well, copy your access token and paste it in the `.env.development` that you created earlier

## Screenshots
**Before**
Screenshot_2020-03-09 Home(1)

**After**
Screenshot_2020-03-26 Home

## Notes
- ~~it won't work on deploy preview~~
- it only returns all events from the Facebook page, not yet tested for upcoming events

Closes #60

### CODE REVIEW

**Comment by galacemiguel** [comment]
Right now the data is hardcoded, is it possible for you to pull the data automatically from the events on our Facebook page? @jromest

**Comment by itsdyeyson** [comment]
facebook graphql api access token added on production environment

File: gatsby-config.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+require("dotenv").config({
+  path: `.env.${process.env.NODE_ENV}`,
+})
+
+const facebookPageID = 431794124327063 // ReactJS PH Facebook Page ID
```
Comment (galacemiguel) [inline gatsby-config.js]: **`NITPICK`** Can you use constant case?

File: src/components/meetupSection.js [hunk 2]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
             key={id}
-            flexDirection="column"
-            width="100%"
-            maxWidth="30em"
-            color="darkblue"
-            mt={2}
-            mx="auto"
+            as="li"
+            bg="white"
+            p={2}
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: Can you use `py="1"` and `px="2"`?

File: src/components/meetupSection.js [hunk 3]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
-            <Flex
-              bg="white"
-              p={2}
-              flexDirection="column"
-              flex={1}
+            <Box
+              as="time"
+              dateTime={start_time}
+              pr={2}
+              mr={2}
```
Comment (galacemiguel) [inline src/components/meetupSection.js]:
```suggestion
              mr={2}
              fontSize={[1, 2]}
```

File: src/components/meetupSection.js [hunk 4]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
-                  <Text ml="0.5rem">›</Text>
-                </Flex>
-                <Text my="0.25rem">{date}</Text>
-                <Text>{venue}</Text>
-              </Box>
+              {formatDate(start_time)}
+            </Box>
+
+            <Box flex={1} pt={[1, 0]} pb={[2, 0]}>
+              <Text
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: Can you add a right margin here just so it doesn't ever push up against the button?

File: src/components/meetupSection.js [hunk 5]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
+                as="h4"
+                fontSize={[2, 3]}
+                fontWeight="heading"
+                sx={{ textTransform: "uppercase" }}
+              >
+                {name}
+              </Text>
+              <Text>{`${place.name}, ${place.location &&
+                place.location.city}`}</Text>
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: image

Can you add some checks to make sure we don't show `null`?

File: src/components/meetupSection.js [hunk 6]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
-                    <Button variant="outline" mt="1">
-                      ▩ SPEAK
-                    </Button>
-                  </>
-                ) : (
-                  <Button variant="outline">◉ Watch</Button>
-                )}
-              </Box>
-            </Flex>
+            <Button
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: Can you add `mr="6px"` and `mb="6px"` to account for the box shadow, so the padding on the container looks right?

File: src/components/meetupSection.js [hunk 7]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
-    </Container>
-  </Box>
-)
+
+        <Box
+          as="hr"
+          my={3}
+          mx="auto"
+          sx={{ borderColor: "white", height: "2px", maxWidth: "170px" }}
+        />
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: No `hr` here please.

File: src/components/meetupSection.js [hunk 8]
```diff
@@ -57,83 +59,95 @@ const MeetupSection = () => (
...
-          gridGap: "3rem 2rem",
-          gridTemplateColumns: [
-            "auto",
-            null,
-            `repeat(auto-fit, minmax(25em, 1fr))`,
-          ],
-        }}
-      >
-        {meetups.map(({ id, logo, title, date, venue, replay }) => (
+        {data.map(({ id, name, start_time, place }) => (
```
Comment (galacemiguel) [inline src/components/meetupSection.js]: Can you have a `map` for upcoming events too?

File: gatsby-config.js (hunk 1 above)
Comment (jromest) [reply gatsby-config.js]: Sure! 😄

File: src/components/meetupSection.js (hunk 8 above)
Comment (jromest) [reply src/components/meetupSection.js]: I haven't tested it yet, I don't know what the data looks like for the upcoming events.

Comment (galacemiguel) [reply src/components/meetupSection.js]: Hmm, can you try creating an upcoming draft event (just don't publish it) so we can test it?

Comment (jromest) [reply src/components/meetupSection.js]: We can test upcoming events if we have an active event.

**Pull request merged**